File: src/main/java/akostenko/aicars/drawing/Scale.java
package akostenko.aicars.drawing;

public class Scale {
    private final float size;
    private final float pixels;

    public Scale(float size, float pixels) {
        this.size = size;
        this.pixels = pixels;
    }

    public float getSize() {
        return size;
    }

    public float getPixels() {
        return pixels;
    }
}

File: src/main/kotlin/akostenko/aicars/drawing/Scale.kt
package akostenko.aicars.drawing

import akostenko.math.vector.Vector

data class Scale(val size: Double, val pixels: Float) {
    fun to(value: Double) : Float = value.toFloat() / size.toFloat() * pixels
    fun to(vector: Vector) : Vector = vector / size * pixels.toDouble()
    fun from(pixels: Float) : Double = pixels / this.pixels * size
}
